From: Ilias Tsitsimpis <i.tsitsimpis@gmail.com>
Date: Sat, 10 Oct 2015 19:55:12 +0300
Subject: Patch Makefiles to honor CFLAGS and LDFLAGS

This patch allows the hardening flags injection into all Makefiles
during build time. It also removes -Os and -O0 from upstream config.mk
to allow DEB_BUILD_OPTIONS=noopt.

Also, remove the -s linker option, as well as any explicit strip
invocation to allow proper working of nostrip value for
DEB_BUILD_OPTIONS.

Author: Vasudev Kamath <kamathvasudev@gmail.com>
Forwarded: no
---
 dmenu/config.mk    | 6 +++---
 lsw/config.mk      | 6 +++---
 slock/config.mk    | 6 +++---
 sprop/config.mk    | 6 +++---
 sselp/config.mk    | 6 +++---
 ssid/Makefile      | 1 -
 ssid/config.mk     | 4 ++--
 swarp/Makefile     | 1 -
 swarp/config.mk    | 4 ++--
 tabbed/config.mk   | 6 +++---
 wmname/Makefile    | 1 -
 wmname/config.mk   | 6 +++---
 xssstate/config.mk | 6 +++---
 13 files changed, 28 insertions(+), 31 deletions(-)

diff --git a/dmenu/config.mk b/dmenu/config.mk
index 50fe52c..a6984f3 100644
--- a/dmenu/config.mk
+++ b/dmenu/config.mk
@@ -23,9 +23,9 @@ INCS = -I${X11INC} -I${FREETYPEINC}
 LIBS = -L${X11LIB} -lX11 ${XINERAMALIBS} ${FREETYPELIBS}
 
 # flags
-CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
-CFLAGS   = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS  = -s ${LIBS}
+CPPFLAGS += -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"${VERSION}\" ${XINERAMAFLAGS}
+CFLAGS   += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS  += ${LIBS}
 
 # compiler and linker
 CC = cc
diff --git a/lsw/config.mk b/lsw/config.mk
index 1d32307..400b2ca 100644
--- a/lsw/config.mk
+++ b/lsw/config.mk
@@ -13,9 +13,9 @@ INCS = -I${X11INC}
 LIBS = -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS   = -ansi -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS  = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS   += -ansi -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS  += ${LIBS}
 
 # compiler and linker
 CC = cc
diff --git a/slock/config.mk b/slock/config.mk
index 74429ae..1c1374a 100644
--- a/slock/config.mk
+++ b/slock/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -lcrypt -L${X11LIB} -lX11 -lXext -lXrandr
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\" -D_DEFAULT_SOURCE -DHAVE_SHADOW_H
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\" -D_DEFAULT_SOURCE -DHAVE_SHADOW_H
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 COMPATSRC = explicit_bzero.c
 
 # On OpenBSD and Darwin remove -lcrypt from LIBS
diff --git a/sprop/config.mk b/sprop/config.mk
index 826cdcb..7da5f46 100644
--- a/sprop/config.mk
+++ b/sprop/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 
 # compiler and linker
 CC = cc
diff --git a/sselp/config.mk b/sselp/config.mk
index ede4ad1..aa54c7e 100644
--- a/sselp/config.mk
+++ b/sselp/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 
 # compiler and linker
 CC = cc
diff --git a/ssid/Makefile b/ssid/Makefile
index 3cfd520..8039ab1 100644
--- a/ssid/Makefile
+++ b/ssid/Makefile
@@ -24,7 +24,6 @@ ${OBJ}: config.mk
 ssid: ${OBJ}
 	@echo LD $@
 	@${LD} -o $@ ${OBJ} ${LDFLAGS}
-	@strip $@
 
 clean:
 	@echo cleaning
diff --git a/ssid/config.mk b/ssid/config.mk
index 0b020f2..b675a4b 100644
--- a/ssid/config.mk
+++ b/ssid/config.mk
@@ -11,8 +11,8 @@ INCS = -I. -I/usr/include
 LIBS = -L/usr/lib -lc
 
 # flags
-CFLAGS = -Os ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS}
+CFLAGS += ${INCS} -DVERSION=\"${VERSION}\" ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 #CFLAGS = -g -Wall -O2 ${INCS} -DVERSION=\"${VERSION}\"
 #LDFLAGS = -g ${LIBS}
 
diff --git a/swarp/Makefile b/swarp/Makefile
index 8e1c848..514855b 100644
--- a/swarp/Makefile
+++ b/swarp/Makefile
@@ -24,7 +24,6 @@ ${OBJ}: config.mk
 swarp: ${OBJ}
 	@echo LD $@
 	@${LD} -o $@ ${OBJ} ${LDFLAGS}
-	@strip $@
 
 clean:
 	@echo cleaning
diff --git a/swarp/config.mk b/swarp/config.mk
index f6d6fd5..b3bae48 100644
--- a/swarp/config.mk
+++ b/swarp/config.mk
@@ -15,8 +15,8 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CFLAGS = -Os ${INCS} -DVERSION=\"${VERSION}\"
-LDFLAGS = ${LIBS}
+CFLAGS += ${INCS} -DVERSION=\"${VERSION}\" ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 #CFLAGS = -g -Wall -O2 ${INCS} -DVERSION=\"${VERSION}\"
 #LDFLAGS = -g ${LIBS}
 
diff --git a/tabbed/config.mk b/tabbed/config.mk
index 5279711..78ce088 100644
--- a/tabbed/config.mk
+++ b/tabbed/config.mk
@@ -12,9 +12,9 @@ INCS = -I. -I/usr/include
 LIBS = -L/usr/lib -lc -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\" -D_BSD_SOURCE
-CFLAGS = -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\" -D_BSD_SOURCE
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 
 # Solaris
 #CFLAGS = -fast ${INCS} -DVERSION=\"${VERSION}\"
diff --git a/wmname/Makefile b/wmname/Makefile
index fc14296..d902583 100644
--- a/wmname/Makefile
+++ b/wmname/Makefile
@@ -23,7 +23,6 @@ ${OBJ}: config.mk
 wmname: ${OBJ}
 	@echo LD $@
 	@${LD} -o $@ ${OBJ} ${LDFLAGS}
-	@strip $@
 
 clean:
 	@echo cleaning
diff --git a/wmname/config.mk b/wmname/config.mk
index ca6d6c4..034ead9 100644
--- a/wmname/config.mk
+++ b/wmname/config.mk
@@ -15,9 +15,9 @@ INCS = -I. -I/usr/include -I${X11INC}
 LIBS = -L/usr/lib -lc -L${X11LIB} -lX11
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
-LDFLAGS = -s ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 
 # compiler and linker
 CC = cc
diff --git a/xssstate/config.mk b/xssstate/config.mk
index 02cf7d4..e4a52b7 100644
--- a/xssstate/config.mk
+++ b/xssstate/config.mk
@@ -12,9 +12,9 @@ INCS = -I. -I/usr/include
 LIBS = -L/usr/lib -lc -lX11 -lXss
 
 # flags
-CPPFLAGS = -DVERSION=\"${VERSION}\"
-CFLAGS = -g -std=c99 -pedantic -Wall -O0 ${INCS} ${CPPFLAGS}
-LDFLAGS = -g ${LIBS}
+CPPFLAGS += -DVERSION=\"${VERSION}\"
+CFLAGS += -g -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += -g ${LIBS}
 
 # Solaris
 #CFLAGS = -fast ${INCS} -DVERSION=\"${VERSION}\"
